pyproject.toml

```diff
@@ -14,12 +15,17 @@ python = "^3.10"
 pydantic = "^1.8.2"
 openai = "^0.28.1"
 cohere = "^4.32"
+numpy = "^1.26.2"
+scipy = "^1.11.4"
```

Comment:
Seems like these libraries were missing, though being used within the project

Reply:
`numpy` should be included for sure — I don't know where `scipy` is being used though?

Reply:
Ok, nvm I got confused, I see we're actually using `numpy.linalg.norm` not the scipy version